clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.cpp

```diff
@@ -899,6 +909,15 @@ Value LowerFunction::rewriteCallOp(const LowerFunctionInfo &CallInfo,
     }
   }();
 
+  for (auto user : Caller.getOperation()->getUsers()) {
```

Comment:
Seems like updating the users of a transformed call should be a common thing to do in target lowering, I wonder if there's a better place for this or if the logic can be incorporated in a pre-existing place. cc @sitio-couto

Reply:
This seems hacky. You should not have to handle more bit casts after the return value lambda, which is type coercion, is already concluded. This is especially not in a path common to all argument types _and_ to all store ops. This should be done _only_ to struct return values and their respective stores.

This should be within the `createCoercedValue` call you left commented above. Please move this logic there.

Reply:
Was this fixed? Seems like I'm seeing the same code?

@bruteforceboy please allow the reviewers to resolve conversations, easier to make sure things got addressed.

Reply:
oh, sorry about that. I moved the rewrite for the store operations, so it is only for struct return values.